rust/src/quic/parser.rs

```diff
@@ -286,7 +288,18 @@ impl QuicHeader {
                             return Err(nom7::Err::Error(QuicError::InvalidPacket));
                         }
                     }
+                } else if version == QuicVersion::V2 {
```

Comment:
branch looks the same as the next, is it needed?

Reply:
Nope cf https://datatracker.ietf.org/doc/rfc9369/ 3.2.  Long Header Packet Types

Quic RFC is there to mess up with developers : there is a shift in the numbers for the quictypes...

(I think clippy catches branches with same code)

Reply:
ah yes I see it now